src/service/performance-impl.js

```diff
@@ -380,15 +380,13 @@ export class Performance {
     const didStartInPrerender = !this.viewer_.hasBeenVisible();
     let docVisibleTime = didStartInPrerender ? -1 : this.initTime_;
 
-    // This is only relevant if the viewer is in prerender mode.
+    // This will only be relevant if the viewer is in prerender mode.
     // (hasn't been visible yet, ever at this point)
-    if (didStartInPrerender) {
-      this.viewer_.whenFirstVisible().then(() => {
-        docVisibleTime = this.win.Date.now();
-        // Mark this first visible instance in the browser timeline.
-        this.mark('visible');
-      });
-    }
+    this.viewer_.whenFirstVisible().then(() => {
+      docVisibleTime = this.win.Date.now();
+      // Mark this first visible instance in the browser timeline.
+      this.mark('visible');
```

Comment:
Ooo, one thing I didn't think about is that this will probably skew the metrics because non-prerendering pages will have 0ms visible ticks. Could we create a new tick for the combination?

Reply:
I am not sure what you mean by a new tick for the combination?  

This was created solely for this new amp work so the only current use case right now is the test cases I have authored. 

Couldn't we just filter out the 0 cases?

Reply:
> I am not sure what you mean by a new tick for the combination?

Leave the old code as is, and add a new:

```js
this.viewer_.whenFirstVisible().then(() => {
      this.mark('visibleOrNoPrerender');
});
```

> Couldn't we just filter out the 0 cases?

I'm sure we could, but other viewers might not have the ability to filter the metrics reports.